app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.js

```diff
@@ -207,7 +209,11 @@ export async function switchChain(
         permissionedChainIds === undefined ||
         !permissionedChainIds.includes(chainId)
       ) {
-        await requestPermittedChainsPermission([chainId]);
+        if (isAddFlow) {
+          await grantPermittedChainsPermissionIncremental([chainId]);
+        } else {
+          await requestPermittedChainsPermission([chainId]);
```

Comment:
Would merging this change have an effect on prod MetaMask (not feature flagged)?

Reply:
It shouldn't. That whole block is guarded by a check for the feature flag 

https://github.com/MetaMask/metamask-extension/blob/144dea496a48a00715de5c4cf16d13df00e43f3f/app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.js#L201